### PR TITLE
A lot of questions. need to go over it again

### PR DESCRIPTION
**Pull request by mikeadeleke**
Submitting a pull request would not even be a good avenue because I do not believe I understood this assignment. However, it gives the ability for you to see what I have so far. What do I need to go over if these are my questions:

1) Refactoring a line to add in a "13.times do" statement
2) Move ranking

### CODE REVIEW

File: superfight.rb [hunk 1]
```diff
@@ -6,6 +6,15 @@
 puts "What is your second fighter's name?"
 fighter_b = $stdin.gets
 
-match = Match.new(Fighter.new(fighter_a), Fighter.new(fighter_b))
+match = Match.new(Fighter.new(fighter_a), Fighter.new(fighter_b)) do
```
Comment (jwo) [inline superfight.rb]: This would be extremely difficult to full off -- you'd need to have a block on initialize ("new").

But, instead, I think you want "tap" here .... think of tap like this:

```
class Thermometer
  attr_accessor :temp
end


gauge = Thermometer.new.tap{|t| t.temp = 95}

# OR

gauge = Thermometer.new.tap do |t| 
  t.temp = 95
end

gauge.temp
=> 95
```

tldr: `match = Match.new(Fighter.new(fighter_a), Fighter.new(fighter_b)).tap |match| do`

**Comment by jwo** [comment]
Not sure what you meant by "Move ranking" -- can you expand?

**Comment by mikeadeleke** [comment]
Almost done on the Panda level but running into an undefined local variable for "fighter_a". 

Yeah I am a little confused with the Tiger level question:
1. Instead of randomly selecting the winner of a turn, use the Move's ranking to determine a winner

**Comment by jwo** [comment]
> Instead of randomly selecting the winner of a turn, use the Move's ranking to determine a winner

In https://github.com/RubyoffRails/Episode2/blob/master/lib/turn.rb there's this code:

``` ruby
def determine_winner
  [@move_a, @move_b].sample
end    
```

This just gets a random move between a and b. Instead, it should determine based on the move's ranking:

``` ruby
class Move

        attr_reader :type, :ranking
        def initialize(type)
                @type = type
                @ranking = rand(100)
        end

end
```

So the ranking is just random. Instead, decide on what `@type` is. Like a strike is 50, a block is 40, a sweep_the_leg is 60. Or something.

**Comment by mikeadeleke** [comment]
Okay things are a little more clear. What would be a good way to generate a random move now?

File: superfight.rb [hunk 2]
```diff
@@ -2,10 +2,50 @@
...
+	def moves
+		strike = 50
+		block = 40
+		leg_sweep = 60
+	end
+end
+
+match = Match.new(Fighter.new(@fighter_a), Fighter.new(@fighter_b)) do
+	3.times.map do
+		match = Match.new
```
Comment (jwo) [inline superfight.rb:51]: A match should contain 13 rounds. Each round has a move by each player

Remember, when you `puts "Fighter A  -- #{@fighter_a.name} -- won"` -- that should be a replay of the game's results. Not looping through and doing it,.

File: superfight.rb [hunk 3]
```diff
@@ -2,10 +2,50 @@
...
+
+	attr_reader :strike, :block, :leg_sweep
+
+	def initialize
+		@strike = strike
+		@block = block
+		@leg_sweep = leg_sweep
+	end
+
+	def moves
```
Comment (jwo) [inline superfight.rb]: try returning a hash here.

File: superfight.rb (hunk 2 above)
Comment (mikeadeleke) [inline superfight.rb:51]: Hmm. what do you mean by replay instead of looping? do you mean that there should be a line where there is a rematch instead of spitting out the same result?

Comment (jwo) [inline superfight.rb:51]: I mean that that I should be able to view the results, and then pass the match object to you, and then you view the same results. 

It means you have to store the rounds in an array for the match.

File: superfight.rb (hunk 3 above)
Comment (mikeadeleke) [inline superfight.rb]: Hmmm. Im not really sure what hash that would be. Should i rewatch the video?

Comment (jwo) [inline superfight.rb]: What hash would you need to display all of the data in your "moves" method?

File: superfight.rb (hunk 2 above)
Comment (mikeadeleke) [inline superfight.rb:51]: I'm thinking, maybe I should make a new method called "rounds" that has an array. I also now  have a "moves" method. I can call "values" on moves and pass in the values. Is that what you were thinking?

File: superfight.rb (hunk 3 above)
Comment (mikeadeleke) [inline superfight.rb]: I put in a test hash but this is something where I really am unsure.

File: superfight.rb [hunk 4]
```diff
@@ -2,10 +2,58 @@
...
+		@leg_sweep = leg_sweep
+	end
+
+	def values
+		strike = 50
+		block = 40
+		leg_sweep = 60
+	end
+
+	def moves
```
Comment (jwo) [inline superfight.rb:40]: I can tell you're lost here. 

So, first question: why are you not using the provided Move class? `lib/move.rb`

second, if you have some possible moves, it should look like

```
def possible_moves
  {strike: 50, block: 40, leg_sweep: 60}
end
```

Later, when you need to randomly pick one of those? A hash can turn into an array

``` ruby
Hash[possible_moves.to_a.sample(1)]
```

Try it out: http://rubyfiddle.com/riddles/d041c
